backdrop/transformers/dispatch.py

```diff
@@ -107,11 +108,18 @@ def get_or_get_and_create_output_dataset(transform, input_dataset):
     )
     output_data_set_config = admin_api.get_data_set(output_group, output_type)
     if not output_data_set_config:
-        data_set_config = dict(input_dataset.items() + {
+        data_set_config = {
             'data_type': output_type,
             'data_group': output_group,
-        }.items())
-        del(data_set_config['name'])
+            'bearer_token': input_dataset['bearer_token'],
+            'realtime': input_dataset['realtime'],
+            'published': input_dataset['published'],
+            'max_age_expected': input_dataset['max_age_expected'],
+        }
+
+        if 'capped_size' in input_dataset and input_dataset['capped_size']:
```

Comment:
It's definitely not possible that input_dataset might also not have a key/value pair for bearer_token/realtime/published/max_age_expected?